brave/src/main/java/brave/propagation/CurrentTraceContext.java

```diff
@@ -175,7 +175,8 @@ public Scope maybeScope(@Nullable TraceContext currentSpan) {
    */
   public static final class Default extends ThreadLocalCurrentTraceContext {
     // Inheritable as Brave 3's ThreadLocalServerClientAndLocalSpanState was inheritable
-    static final InheritableThreadLocal<TraceContext> INHERITABLE = new InheritableThreadLocal<>();
+    static final InheritableThreadLocal<Object[]> INHERITABLE =
```

Comment:
one way to deal with this is to special-case an inheritable implementation. meaning if the thread local is inheritable, use normal non-wrapper approach